plugins/catalog-backend/src/ingestion/processors/results.ts

```diff
@@ -67,3 +71,7 @@ export function entity(
 ): CatalogProcessorResult {
   return { type: 'entity', location: atLocation, entity: newEntity };
 }
+
+export function relation(spec: EntityRelationSpec): CatalogProcessorResult {
```

Comment:
i've been thinking that from the processor POV, it would probably be nicer if the interface was on the form `emit.relation(spec)`, `emit.entity(spec)` etc so it's more completion friendly and you don't have to do that extra `results` import. Could even make it `emit.outgoingRelation` and `emit.incomingRelation` to not have to specify the "self" part (the calling code knows what that is already)

Reply:
Yup, not now though ofc. Another option to the two relation emit forms is to have a bidirectional one where you specify the type in each direction. Tbh I don't think it's a lot of typing and more clear to just have the single relation emit at this point tho.